analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -360,10 +362,11 @@ open class Npm(
         return Pair(id.toCoordinates(), module)
     }
 
-    protected open fun getRemotePackageDetails(workingDir: File, packageName: String): JsonNode {
-        val process = run(workingDir, "view", "--json", packageName)
-        return jsonMapper.readTree(process.stdout)
-    }
+    protected open fun getRemotePackageDetails(workingDir: File, packageName: String): JsonNode =
+        npmViewCache.getOrPut(packageName) {
```

Comment:
I am also not sure about the semantics of `getOrPut` regarding concurrent access. `ConcurrentMap` defines special atomic update functions. Maybe the Kotlin extension is aware of this, but I do not for sure.

Reply:
> Maybe the Kotlin extension is aware of this, but I do not for sure.

Yes it is, there's a special `fun ConcurrentMap<K, V>.getOrPut()`.

Reply:
Yes, the `getOrPut` implementation for `ConcurrentHashMap` uses `putIfAbsent`.